### PR TITLE
Added tests for elasticsearch

### PR DESCRIPTION
**Pull request by sushma1118**
* Cluster-setup, setup and functional tests for elasticserach are added.

### CODE REVIEW

File: Openshift-EE/pipelines/ElasticSearch/stages/1-cluster-setup/HUNE-cluster-state/cluster-state [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+if [ "$ready_nodes" -eq 6 ]; then
+echo "Number of nodes in ready state is $ready_nodes"
+echo "******Cluster is in Healthy state****"
+
+echo "*************Dumping cluster state********"
+sshpass -p $pass ssh -o StrictHostKeyChecking=no $user@$ip -p 1658 kubectl get nodes
+sshpass -p $pass ssh -o StrictHostKeyChecking=no $user@$ip -p 1658 kubectl get pod --all-namespaces
+
+echo "cloning e2e-openshift repo*************"
+sshpass -p $pass ssh -o StrictHostKeyChecking=no $user@$ip -p $port 'git clone -b OpenEBS-base https://github.com/openebs/e2e-openshift.git'
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/ElasticSearch/stages/1-cluster-setup/HUNE-cluster-state/cluster-state]: @sushma1118 this test scripts is for elasticsearch pipeline. Clone the elastic search branch.

File: Openshift-EE/pipelines/ElasticSearch/stages/1-cluster-setup/HUNE-cluster-state/cluster-state [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+sshpass -p $pass ssh -o StrictHostKeyChecking=no $user@$ip -p $port 'python3 e2e-openshift/Openshift-EE/utils/result/result_update.py '"'$CI_JOB_ID'"' HUNE 1-cluster-setup "Configure OpenShift cluster" Pass '"'$CI_PIPELINE_ID'"' '"'$current_time'"' '"'$CI_COMMIT_SHA'"' '"'$gittoken'"''
+else
+echo "All nodes are not ready"
+echo "******Cluster is in Unhealthy state*******"
+echo "cloning e2e-openshift repo*************"
+sshpass -p $pass ssh -o StrictHostKeyChecking=no $user@$ip -p $port 'git clone -b OpenEBS-base https://github.com/openebs/e2e-openshift.git'
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/ElasticSearch/stages/1-cluster-setup/HUNE-cluster-state/cluster-state]: @sushma1118  this test scripts is for elasticsearch pipeline. Clone the elastic search branch.

File: Openshift-EE/pipelines/ElasticSearch/stages/2-setup/12PS-storage-policies/openebs-sc [hunk 3]
```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+set -x
+pod() {
+sshpass -p $pass ssh -o StrictHostKeyChecking=no $user@$ip -p $port 'git clone -b OpenEBS-base https://github.com/openebs/e2e-openshift.git'
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/ElasticSearch/stages/2-setup/12PS-storage-policies/openebs-sc]: @sushma1118 this test scripts is for elasticsearch pipeline. Clone the elastic search branch.

File: Openshift-EE/pipelines/ElasticSearch/stages/2-setup/12PS-storage-policies/openebs-sc [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+sshpass -p $pass ssh -o StrictHostKeyChecking=no $user@$ip -p $port 'cd e2e-openshift && bash Openshift-EE/pipelines/OpenEBS-base/stages/2-setup/1CXH-storage-policies/openebs-sc node '"'$CI_JOB_ID'"'' '"'$CI_PIPELINE_ID'"' '"'$CI_COMMIT_SHA'"'
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/ElasticSearch/stages/2-setup/12PS-storage-policies/openebs-sc]: @sushma1118 Is this test case ID is correct?

File: Openshift-EE/pipelines/ElasticSearch/stages/2-setup/12PS-storage-policies/openebs-sc [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+node() {
+
+time="date"
+current_time=$(eval $time)
+job_id=$(echo $1)
+pipeline_id=$(echo $2)
+commit_id=$(echo $3)
+case_id=1CXH
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/ElasticSearch/stages/2-setup/12PS-storage-policies/openebs-sc]: @sushma1118 check the case ID here also?

File: Openshift-EE/pipelines/ElasticSearch/stages/3-functional/5Q19-deploy-elasticsearch/deploy-elasticsearch [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+present_dir=$(pwd)
+echo $present_dir
+
+bash Openshift-EE/utils/e2e-cr jobname:app-replica-scaleup-jiva jobphase:Waiting init_time:"$current_time" jobid:"$job_id" pipelineid:"$pipeline_id" testcaseid:"$case_id"
+bash Openshift-EE/utils/e2e-cr jobname:app-replica-scaleup-jiva jobphase:Running
+
+
+echo "*******Deploying Elasticsearch Application****"
+
+test_name=$(bash Openshift-EE/utils/generate_test_name testcase=busybox-provision-target-affinity metadata="")
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/ElasticSearch/stages/3-functional/5Q19-deploy-elasticsearch/deploy-elasticsearch]: @sushma1118 Is this test for deploy elasticsearch or busybox?

File: Openshift-EE/pipelines/ElasticSearch/stages/3-functional/5Q19-deploy-elasticsearch/deploy-elasticsearch [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+commit_id=$(echo $3)
+case_id=5Q19
+
+time="date"
+current_time=$(eval $time)
+
+present_dir=$(pwd)
+echo $present_dir
+
+bash Openshift-EE/utils/e2e-cr jobname:app-replica-scaleup-jiva jobphase:Waiting init_time:"$current_time" jobid:"$job_id" pipelineid:"$pipeline_id" testcaseid:"$case_id"
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/ElasticSearch/stages/3-functional/5Q19-deploy-elasticsearch/deploy-elasticsearch]: @sushma1118 Change the job name to deploy-elasticsearch or elasticsearch-deploy

File: Openshift-EE/pipelines/ElasticSearch/stages/3-functional/5Q19-deploy-elasticsearch/deploy-elasticsearch [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+ | pvcname           | env    | openebs-busybox                     | openebs-esearch                                  |
+ | Litmus Job name   | name   | generateName: litmus-busybox-deploy | generateName: busybox-provision-app-replica      |
+ | storage class     | env    | openebs-cstor-sparse                | openebs-cstor-sparse                             |
+ | Action            | env    | value: provision                    | value: provision                                 |
+   ----------------------------------------------------------------------------------------------------------------------
+EOF
+
+
+sed -i -e 's/imagePullPolicy: Always/imagePullPolicy: IfNotPresent/g' \
+-e 's/app: busybox-litmus/app: elasticsearch-deployment-litmus/g' \
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/ElasticSearch/stages/3-functional/5Q19-deploy-elasticsearch/deploy-elasticsearch]: @sushma1118 check the `sed` operations. In elasticsearch deployment, there will not be any busybox related labels and namespaces will not be there to replace.

File: Openshift-EE/pipelines/ElasticSearch/stages/3-functional/5Q19-deploy-elasticsearch/deploy-elasticsearch [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+cat deploy_affinity_test.yml
+
+bash ../Openshift-EE/utils/litmus_job_runner label='app:elasticsearch-deployment-litmus' job=deploy_affinity_test.yml
+cd ..
+bash Openshift-EE/utils/dump_cluster_state;
+bash Openshift-EE/utils/event_updater jobname:elasticsearch-provision $test_name jobid:"$job_id" pipelineid:"$pipeline_id" testcaseid:"$case_id"
+
+if [ "$?" != "0" ]; then
+python3 Openshift-EE/utils/result/result_update.py $job_id 5Q19 3-functional "Checking Application and target scheduled on same Node" Fail $pipeline_id "$current_time" $commit_id $gittoken
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/ElasticSearch/stages/3-functional/5Q19-deploy-elasticsearch/deploy-elasticsearch]: @sushma1118 change the job description.

File: Openshift-EE/pipelines/ElasticSearch/stages/3-functional/86TL-App-replica-scaleup/app-replica-scaleup [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+commit_id=$(echo $3)
+case_id=86TL
+
+time="date"
+current_time=$(eval $time)
+
+present_dir=$(pwd)
+echo $present_dir
+
+bash Openshift-EE/utils/e2e-cr jobname:app-replica-scaleup-jiva jobphase:Waiting init_time:"$current_time" jobid:"$job_id" pipelineid:"$pipeline_id" testcaseid:"$case_id"
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/ElasticSearch/stages/3-functional/86TL-App-replica-scaleup/app-replica-scaleup]: @sushma1118 Is this test related to cstor or jiva

File: Openshift-EE/pipelines/ElasticSearch/stages/3-functional/86TL-App-replica-scaleup/app-replica-scaleup [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+  ---------------------------------------------------------------------------------------------------------------------|
+ | appLabel          | env    | app=cassandra                       | app=elasticsearch                                | 
+ | appNamespace      | env    | app-cass-ns                         | app-esearch-ns                                   | 
+ | Litmus job label  | label  | app: cassandra-scaleup-litmus       | app:  elasticsearch-deployment-litmus            |                         |
+  ----------------------------------------------------------------------------------------------------------------------
+EOF
+
+sed -i -e 's/app: cassandra-scaleup-litmus/app: elasticsearch-deployment-litmus /g' \
+-e 's/value: '\''app=cassandra'\''/value: '\''app=elasticsearch'\''/g' \
+-e 's/value: app-cass-ns/value: app-esearch-ns/g' run_scaleup_test.yml
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/ElasticSearch/stages/3-functional/86TL-App-replica-scaleup/app-replica-scaleup]: @sushma1118 Is this file name `run_scaleup_test.yml` or `run_scaleup_test_jiva.yml`

File: Openshift-EE/pipelines/ElasticSearch/stages/3-functional/86TL-App-replica-scaleup/app-replica-scaleup [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+echo "Running the litmus test for Elasticsearch Deployment Scaleup.."
+cat run_scaleup_test.yml
+
+bash ../Openshift-EE/utils/litmus_job_runner label='app: elasticsearch-deployment-litmus' job=run_scaleup_test.yml
+cd ..
+bash Openshift-EE/utils/dump_cluster_state;
+bash Openshift-EE/utils/event_updater jobname:app-replica-scaleup $test_name jobid:"$job_id" pipelineid:"$pipeline_id" testcaseid:"$case_id"
+
+if [ "$?" != "0" ]; then
+python3 Openshift-EE/utils/result/result_update.py $job_id 86TL 3-functional "Scale the statefulset application replicas and verify the ring configuration" Fail $pipeline_id "$current_time" $commit_id $gittoken
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/ElasticSearch/stages/3-functional/86TL-App-replica-scaleup/app-replica-scaleup]: @sushma1118 check the job description what was the `ring`

Comment (sushma1118) [reply Openshift-EE/pipelines/ElasticSearch/stages/3-functional/86TL-App-replica-scaleup/app-replica-scaleup]: can you please elaborate it

File: Openshift-EE/pipelines/ElasticSearch/stages/3-functional/5Q19-deploy-elasticsearch/deploy-elasticsearch [hunk 13]
```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+set -x
+
+ pod() {
+echo $CI_JOB_ID
+###clone e2e-openshift-repo
+echo "cloning e2e-openshift repo*************"
+sshpass -p $pass ssh -o StrictHostKeyChecking=no $user@$ip -p 1658 'git clone https://github.com/openebs/e2e-openshift.git'
+sshpass -p $pass ssh -o StrictHostKeyChecking=no $user@$ip -p 1658 'cd e2e-openshift && git checkout elasticsearch && bash Openshift-EE/pipelines/ElasticSearch/stages/3-functional/ELHN-deploy-prometheus/prometheus-provision node '"'$CI_JOB_ID'"''
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/ElasticSearch/stages/3-functional/5Q19-deploy-elasticsearch/deploy-elasticsearch]: @sushma1118 check the test script directory.

File: Openshift-EE/pipelines/ElasticSearch/stages/3-functional/5Q19-deploy-elasticsearch/deploy-elasticsearch [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+
+ kubectl apply -f Openshift-EE/utils/result-elasticsearch-provision.yml
+
+ test_name=$(bash Openshift-EE/utils/generate_test_name testcase=elasticsearch-deployment metadata="")
+echo $test_name
+
+ cd litmus
+
+ echo "Running the litmus test.."
+run_test=apps/prometheus/deployers/run_litmus_test.yml
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/ElasticSearch/stages/3-functional/5Q19-deploy-elasticsearch/deploy-elasticsearch]: @sushma1118 get the litmus book for elasticsearch deployment.

**Pull request merged**